### PR TITLE
[develop < T0020-GA] Fix label inheritance

### PR DESCRIPTION
**Pull request by g-despot**
- [x] Fixed behavior when labels aren't inherited from parent classes.
- [x] Added tests to cover most inheritance cases.

### CODE REVIEW

**Review by katarinasupe — APPROVED** [review]
Great job on testing!

**Comment by katarinasupe** [comment]
I just reformatted one line with black, and tests should pass now.

**Review by katarinasupe — APPROVED** [review]

File: gqlalchemy/models.py [hunk 1]
```diff
@@ -350,16 +350,22 @@ def field_in_superclass(field, constraint):
 
             return None
 
+        def get_base_labels() -> Set[str]:
+            base_labels = set()
+            nonlocal bases
+            for base in bases:
+                base_labels = base_labels.union(base.labels if hasattr(base, "labels") else set())
```
Comment (BorisTasevski) [inline gqlalchemy/models.py:357]: feels like this union operation could be skipped if `hasattr` returns `False`

**Review by BorisTasevski — COMMENTED** [review]
Maybe a minor improvement.

File: gqlalchemy/models.py (hunk 1 above)
Comment (g-despot) [reply gqlalchemy/models.py]: Thank you for the suggestion. I skipped the union operation if `hasattr` returns `False`.

File: gqlalchemy/models.py [hunk 2]
```diff
@@ -354,7 +354,7 @@ def get_base_labels() -> Set[str]:
             base_labels = set()
             nonlocal bases
             for base in bases:
-                base_labels = base_labels.union(base.labels if hasattr(base, "labels") else set())
+                base_labels = base_labels.union(base.labels) if hasattr(base, "labels") else base_labels
```
Comment (BorisTasevski) [inline gqlalchemy/models.py:357]: `if hasattr(base, "labels"): base_labels.union(base.labels)` ?

Comment (g-despot) [reply gqlalchemy/models.py]: Fixed

File: gqlalchemy/models.py [hunk 3]
```diff
@@ -350,16 +350,23 @@ def field_in_superclass(field, constraint):
...
+            nonlocal bases
+            for base in bases:
+                if hasattr(base, "labels"):
+                    base_labels = base_labels.union(base.labels)
+            return base_labels
+
         cls = super().__new__(mcs, name, bases, namespace, **kwargs)
+
         cls.label = kwargs.get("label", name)
         if name == "Node":
```
Comment (BorisTasevski) [inline gqlalchemy/models.py:364]:
```
if name != "Node":
     cls.labels = {cls.label} | get_base_labels() | kwargs.get("labels", set())
```

?

**Review by BorisTasevski — CHANGES_REQUESTED** [review]
There are conflicts as well

File: gqlalchemy/models.py (hunk 3 above)
Comment (BorisTasevski) [reply gqlalchemy/models.py:364]: And another Question, `|` operator is a shorthand for `union()` method, so why are we using both?

Comment (g-despot) [reply gqlalchemy/models.py]: Thank you for the suggestion, I replaced the block with one if statement and replaced the `|` operator with the `union()` method.

File: gqlalchemy/models.py [hunk 4]
```diff
@@ -356,17 +356,13 @@ def get_base_labels() -> Set[str]:
...
-
         cls.label = kwargs.get("label", name)
-        if name == "Node":
-            pass
-        elif "labels" in kwargs:
-            cls.labels = {cls.label} | kwargs["labels"] | get_base_labels()
-        else:
-            cls.labels = {cls.label} | get_base_labels()
+        if name != "Node":
+            cls.labels = get_base_labels().union({cls.label}, kwargs.get("labels", set()))
```
Comment (katarinasupe) [inline gqlalchemy/models.py:371]: I like the way you changed this 😄

Comment (BorisTasevski) [reply gqlalchemy/models.py:371]: Thanks :)

**Review by BorisTasevski — APPROVED** [review]

**Pull request merged**